bazel/bazel.go

```diff
@@ -23,13 +23,66 @@ import (
 	"io"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"strings"
 
 	blaze_query "github.com/bazelbuild/bazel-watcher/third_party/bazel/master/src/main/protobuf"
 	"github.com/golang/protobuf/proto"
 )
 
-var bazelPath = flag.String("bazel_path", "bazel", "Path to the bazel binary to use for actions")
+var bazelPathFlag = flag.String("bazel_path", "", "Path to the bazel binary to use for actions")
+
+// bazelNpmPath looks up a relative path to a binary from @bazel/bazel
+// This is used as an alternate resolution when no bazel binary is in the $PATH
+// When running from the @bazel/ibazel npm package, our binary is
+// /DIR/node_modules/@bazel/ibazel/bin/$ARCH/ibazel
+// We can find bazel in
+// /DIR/node_modules/@bazel/bazel-$ARCH/bazel-0.28.0-linux-x86_64
+func bazelNpmPath(ibazelBinPath string) (string, error) {
+	s := strings.Split(ibazelBinPath, "/")
+	for i := 0; i + 4 < len(s); i++ {
```

Comment:
No action required, but since you know a lot about compilers you might have a neat trick here. The two are very definitely equivalent, Why `i + 4 < len(s)` instead of `i < len(s) - 4`?

Do compilers do anything where they detect that the inside of the loop isn't modifying either side of the operation, precomputing the rhs of the inequality and then having a const `JNLE` instruction? You probably don't know anything about the GoLang compiler in particular, but I'm curious none the less.

Reply:
I just did it this way since the expression `i+4` appears below, I know that's the index I want to exist, so this way the expression in the loop condition is identical